server/chat.ts

```diff
@@ -1108,13 +1108,19 @@ export class CommandContext extends MessageContext {
 					}
 				}
 				if (tagName === 'button') {
-					if (
-						(this.room.isPersonal || this.room.isPrivate === true) &&
-						!this.user.can('lock') && /<button[^>]/.test(htmlContent.toLowerCase().replace(/\s*style\s*=\s*"?[^"]*"\s*>/g, '>'))
-					) {
-						this.errorReply(`You do not have permission to use scripted buttons (buttons with attributes other than style="...") in HTML.`);
-						this.errorReply(`If you just want to link to a room, you can do this: <a href="/roomid"><button>button contents</button></a>`);
-						return null;
+					if (tagContent.replace(/\s*(?:name\s*=\s*"\s*msg\s*"|(?:value|style)\s*=\s*"[^"]*")/gi, '') !== "button") {
```

Comment:
I don't think this is accounting for word boundaries correctly.

```suggestion
					if (tagContent.replace(/\b(?:name\s*=\s*"msg"|(?:value|style)\s*=\s*"[^"]*")/gi, '').trim() !== "button") {
```